python/src/iceberg/expressions/base.py

```diff
@@ -267,6 +227,31 @@ def __str__(self) -> str:
         return "false"
 
 
+@dataclass
+class UnboundIn(BooleanExpression, UnboundPredicate):
```

Comment:
You probably want to use `In` for `UnboundIn` and `BoundIn` for the bound version. Users interact the most with the unbound versions and don't need to know much about expression binding.

Reply:
updated this